File: src/main/java/demo01_sentence_segment/SegmentDemo.java
package demo01_sentence_segment;

import com.hankcs.hanlp.dictionary.CustomDictionary;
import com.hankcs.hanlp.dictionary.stopword.CoreStopWordDictionary;
import com.hankcs.hanlp.seg.common.Term;
import com.hankcs.hanlp.tokenizer.NLPTokenizer;
import com.hankcs.hanlp.tokenizer.SpeedTokenizer;
import com.hankcs.hanlp.tokenizer.StandardTokenizer;

import java.util.List;

/**
 * NLP和Standard的都可以，分词效果都可以
 * HighSpeed的分词效果比较差；
 * Created by fansy on 2017/8/24.
 */
enum SegmentType{
    Standard,HighSpeed,NLP
}
public class SegmentDemo {

    public static void main(String[] args) {
        String[]sentenses = {
            "你好，欢迎使用HanLP汉语处理包！",
                "江西鄱阳湖干枯，中国最大淡水湖变成大草原",
                "中国科学院计算技术研究所的宗成庆教授正在教授自然语言处理课程",
                "商品和服务"
        };

        for(String sentense :sentenses){

            System.out.println("NLP:"+process(sentense,SegmentType.NLP));
            System.out.println("Standard:"+process(sentense,SegmentType.Standard));
            System.out.println("Speed:"+process(sentense,SegmentType.HighSpeed));
            System.out.println();
        }

        for(String sentense :sentenses){
            CustomDictionary.add("自然语言处理");
            System.out.println("NLP:"+process(sentense,SegmentType.NLP));
            System.out.println("Standard:"+process(sentense,SegmentType.Standard));
            System.out.println("Speed:"+process(sentense,SegmentType.HighSpeed));
            System.out.println();
        }

        CustomDictionary.remove("自然语言处理");
        for(String sentense :sentenses){

            System.out.println("NLP:"+process(sentense,SegmentType.NLP));
            System.out.println("Standard:"+process(sentense,SegmentType.Standard));
            System.out.println("Speed:"+process(sentense,SegmentType.HighSpeed));
            System.out.println();
        }
    }

    private static String process(String sentense,SegmentType segmentType){
        List<Term> list;
        switch (segmentType){
            case HighSpeed:
                list= SpeedTokenizer.segment(sentense);
                CoreStopWordDictionary.apply(list);
                return list.toString();
            case NLP:
                list =NLPTokenizer.segment(sentense);
                CoreStopWordDictionary.apply(list);
                return list.toString();
            case Standard:
                list = StandardTokenizer.segment(sentense);
                CoreStopWordDictionary.apply(list);
                return list.toString();
            default:
                return null;
        }
    }
}

File: src/main/scala/demo01/SegmentDemo.scala
package demo01

import com.hankcs.hanlp.dictionary.CustomDictionary
import com.hankcs.hanlp.dictionary.stopword.CoreStopWordDictionary
import com.hankcs.hanlp.tokenizer.StandardTokenizer
import scala.collection.JavaConversions._
/**
 * Scala 分词测试
 * Created by fansy on 2017/8/25.
 */
object SegmentDemo {
  def main(args: Array[String]) {
    val sentense = "41,【 日  期 】19960104 【 版  号 】1 【 标  题 】合巢芜高速公路巢芜段竣工 【 作  者 】彭建中 【 正  文 】     安徽合（肥）巢（湖）芜（湖）高速公路巢芜段日前竣工通车并投入营运。合巢芜 高速公路是国家规划的京福综合运输网的重要干线路段，是交通部确定１９９５年建成 的全国１０条重点公路之一。该条高速公路正线长８８公里。（彭建中）"
    CustomDictionary.add("日  期")
    CustomDictionary.add("版  号")
    CustomDictionary.add("标  题")
    CustomDictionary.add("作  者")
    CustomDictionary.add("正  文")
    val list = StandardTokenizer.segment(sentense)
    CoreStopWordDictionary.apply(list)
    println(list.map(x => x.word.replaceAll(" ","")).mkString(","))
  }
}
